### DISCUSSION TITLE
A --file option to attach multiple files

### DISCUSSION THREAD

**Discussion by joeytwiddle** [TheR1D/shell_gpt#98] [Ideas]
I imagined a prompt like this:

```sh
# Proposal
$ sgpt 'Compare' --file 1.csv 'and' --file 2.csv 'and summarise what the changes indicate'
```

Of course, it is possible to do:

```sh
# Existing
$ sgpt "Compare $(cat 1.csv) and $(cat 2.csv) and summarise what the changes indicate"
```
But the proposal has some advantages:

1. We could include a clear header and footer, including the filename, when passing the file to GPT. The header and footer will cleanly separate data from query, and the filename may help GPT to understand the format of the file, as well as providing extra context for the query. (E.g. is this economic data, or medical or weather?)
2. The `--file` option is more standardised than `$(...)` which may not work with all shells.

Disadvantages:

1. To implement, we would need to parse multiple prompts and multiple `--file` options, and retain their ordering. This might not be possible with the current `typer` library.
2. Arguably the second (existing) command is easier for humans to read. The user can also choose exactly how to structure the header/footer/context.

I'm not desperate for this feature, but just had the idea and wanted to share.

**Comment by lpurdy01** [comment]
I think the concept of establishing standard ways for ChatGPT to recognize different types of input and output data types is a great idea! 

I'm already experimenting with it in #108. Where I am attempting to teach it that shell commands between ``` will be executed and the results returned in the next query. like ```ls```. 

There are some challenges in teaching it though, and I think I need some more prompt engineering / auto generated conversation lead in. The `system` role command does make progress, but not enough for it to understand the scope of its interface. 

I have also been looking into LangChain, and they integrated [Unstructured IO](https://github.com/Unstructured-IO/unstructured) to allow for the ingestion of arbitrary file types. I think this could be really powerful and enable shell-gpt to support a wide array of files for context, editing, summarization / memory space compression, and review.

**Comment by TheR1D** [comment]
I think similar results could be achieved using pipes:
```shell
cat a.txt b.txt | sgpt "..."
```
